vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -0,0 +1,159 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <utility>
+#include <vector>
+
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/media/helper.h"
+#include "bat/ledger/internal/media/twitter.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace braveledger_media {
+
+MediaTwitter::MediaTwitter(bat_ledger::LedgerImpl* ledger):
+  ledger_(ledger) {
+}
+
+MediaTwitter::~MediaTwitter() {
+}
+
+// static
+std::string MediaTwitter::GetProfileURL(const std::string& screen_name) {
+  if (screen_name.empty()) {
+    return std::string();
+  }
+
+  return base::StringPrintf("https://twitter.com/%s/", screen_name.c_str());
```

Comment:
hopefully this (and `MediaTwitter::GetProfileImageURL` result) are URL-encoded somewhere before they are fetched?

Reply:
Our `FetchFavIcon` function uses a `BitmapFetcherService` to request these images and both `FetchFavIcon` and `RequestImage` ensure that the passed URL is valid:

https://github.com/brave/brave-core/blob/4c3c61d335847fa337288d208b45d440cb0de2e5/components/brave_rewards/browser/rewards_service_impl.cc#L2034

https://cs.chromium.org/chromium/src/chrome/browser/bitmap_fetcher/bitmap_fetcher_service.cc?sq=package:chromium&dr=CSs&g=0&l=102

cc: @NejcZdovc